### DISCUSSION TITLE
Contributing - How are component styles decided upon?

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/chakra-ui#9305] [General] [answered]
For someone who would like to contribute, what is the process for defining styles? As in, I'm confident I can take components from Ark and do 90% of the work to get them into Chakra. But i'm a little confused as to who or what ensures the styles meet the overall look and feel of Chakra?  Is it just a matter of comparing against existing recipes to sure I would be using the correct tokens etc?

**Comment by segunadebayo** [comment, marked as answer]
Indeed. It mainly takes existing components from Ark and applies consistent theming styles.

- Use intuitive names for recipe variants
- Provide some variants and sizes that align with the naming conventions
- Use the `colorPalette` concept most of the time

I'll look it over and refine it if needed. You don't have to go 100% of the way.
